partitions.go

```diff
@@ -117,6 +117,10 @@ func (ps PartitionSpec) Equals(other PartitionSpec) bool {
 	return ps.id == other.id && slices.Equal(ps.fields, other.fields)
 }
 
+func (ps *PartitionSpec) Fields() []PartitionField {
+	return ps.fields
+}
```

Comment:
the problem with doing this is that it allows a consumer to directly modify partition field values despite us wanting to ensure they are immutable. As a result, a user could change the ID of a partition field and break the `sourceIdToField` map. That's why we didn't have this method already.

Reply:
I have updated the code to clone the slice before returning, this should prevent consumers from directly modifying the fields.